src/elixirHoverProvider.ts

```diff
@@ -0,0 +1,21 @@
+import * as vscode from 'vscode';
+import { ElixirServer } from './elixirServer';
+
+export class ElixirHoverProvider implements vscode.HoverProvider {
+
+  constructor(private server: ElixirServer) { }
+
+  provideHover(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Thenable<vscode.Hover> {
+    return new Promise((resolve, reject) => {
+      this.server.getDocumentation(document, position, (hover: vscode.Hover) => {
+        if (!token.isCancellationRequested) {
+          resolve(hover);
+        } else {
+          console.error('rejecting');
```

Comment:
Is this a meaning full problem if this gets rejected? If yes, maybe we could make the log message more meaningful if not why log at all?